### ISSUE TITLE
SA1102 reports lines with comments. Code fix removes them.

### ISSUE THREAD

**Issue by otac0n** [DotNetAnalyzers/StyleCopAnalyzers#1956]
![sa1102](https://cloud.githubusercontent.com/assets/193584/11760669/23db414e-a058-11e5-9062-b76c027956ab.png)

**Comment by vweijsters** [comment]
I think this is not a bug. StyleCop Classic flags this with SA1102 as well.
This could be an enhancement, but it will need discussion.

**Comment by sharwell** [comment]
@vweijsters The bug here is that the code fix removes the comment, not that a violation is reported.

**Comment by vweijsters** [comment]
Interesting. Under what conditions should the code fix not be offered? I'm guessing when anything but whitespace trivia and end-of-line trivia is present, but I'm not sure.

**Comment by vweijsters** [comment]
After some further thought I think this is not a bug.

The comment is correctly flagged as a violation and therefore it is logical that the code fix would remove it. If the comment should be preserved, the rule should be changed to not flag the comment at all in my opinion.

**Comment by otac0n** [comment]
I was not expecting this behavior when I ran the "fix all occurrences in solution" from another file. I only caught this by being vigilant when staging changes (I do it hunk-by-hunk, typically).

I can see this causing some real grief.

**Comment by vweijsters** [comment]
Fair point. I think it would make sense to not produce violations for comments at all.

**Comment by sharwell** [comment]
Another option would be changing `//` to `////`. And that makes complete sense for *this* example... not sure about other cases though.
